packages/eslint-plugin/lib/config/storybook.ts

```diff
@@ -1,30 +1,29 @@
 // https://github.com/storybookjs/eslint-plugin-storybook/blob/main/lib/configs/recommended.ts
 
-import { mainStorybookFiles, storybookFiles } from "../utils/patterns";
-
 import type { Linter } from "eslint";
+import * as storybookPlugin from "eslint-plugin-storybook";
+import { mainStorybookFiles, storybookFiles } from "../utils/patterns.ts";
 
-const config: Linter.Config = {
-    overrides: [
-        {
-            files: storybookFiles,
-            plugins: ["storybook"],
-            extends: [
-                "plugin:storybook/recommended",
-                "plugin:storybook/csf",
-                "plugin:storybook/csf-strict"
-            ]
-        },
-        {
-            files: mainStorybookFiles,
-            rules: {
-                "storybook/no-uninstalled-addons": "warn"
-            }
+const config: Linter.FlatConfig[] = [
+    {
+        ignores: ["!.storybook"]
+    },
+    {
+        files: storybookFiles,
+        plugins: { "storybook": storybookPlugin },
+        rules: {
+            ...storybookPlugin.configs.recommended.overrides[0].rules,
+            ...storybookPlugin.configs.csf.overrides[0].rules,
+            ...storybookPlugin.configs["csf-strict"].rules
+        }
+    },
+    {
+        files: mainStorybookFiles,
+        plugins: { "storybook": storybookPlugin },
```

Comment:
Looking at the previous config, it seems like this section didn't have the storybook plugin configured?

Reply:
All this is doing is providing the rule definitions. I'm not clear on if this is necessary since it's also provided above. I believe it is, since it's for different file patterns. But even if it isn't necessary, it isn't hurting anything being there.

Reply:
Well, if it's not necessary it kinds of gives the wrong impression to the next maintainer because he will initially believe that is it necessary, ideally, only what is required would be part of the config!

Reply:
The thing I'm not clear on is if it needs to be provided twice. I think it does since the file patterns are different, so ESLint needs to know these rules apply to both file globs. At worst it just overwrites itself. None of this would be necessary once the Storybook plugin updates for flat config. I have to do this because it's not in the right format. Ideally, this would just be part of the Storybook plugin's recommended config and you wouldn't have to manually provide it. So I agree with you there, this shouldn't be necessary, but for now, it is.